src/main/kotlin/com/github/jengelman/gradle/plugins/shadow/internal/WithPatternFilterable.kt

```diff
@@ -0,0 +1,98 @@
+package com.github.jengelman.gradle.plugins.shadow.internal
+
+import groovy.lang.Closure
+import org.gradle.api.GradleException
+import org.gradle.api.file.FileTreeElement
+import org.gradle.api.specs.Spec
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.Internal
+import org.gradle.api.tasks.util.PatternFilterable
+import org.gradle.api.tasks.util.PatternSet
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+
+/**
+ * Helper class for the boilerplate overrides, holding a [PatternSet] exposed as a
+ * [PatternFilterable].
+ */
+public open class WithPatternFilterable(
+  @get:Internal protected val canBeEmpty: Boolean = false,
+  @get:Internal protected val defaultIncludes: Set<String> = setOf(),
+) : PatternFilterable {
+  @get:Internal internal val patternSet: PatternSet = PatternSet()
```

Comment:
Can be resued like https://github.com/GradleUp/shadow/blob/05a47b55ba420faea46e3d1ff61c9d39a3b58700/src/main/kotlin/com/github/jengelman/gradle/plugins/shadow/transformers/ServiceFileTransformer.kt#L32

Reply:
Ah! TIL :)